src/main/java/com/uid2/admin/vertx/service/ServiceService.java

```diff
@@ -169,26 +176,72 @@ private void handleServiceRoles(RoutingContext rc) {
                 return;
             }
 
-            final Set<Role> roles;
-            try {
-                roles = rolesSpec.stream().map(s -> Role.valueOf((String) s)).collect(Collectors.toSet());
-            } catch (IllegalArgumentException e) {
-                ResponseUtil.error(rc, 400, "invalid parameter: roles");
-                return;
+            if (rolesSpec != null) {
+                final Set<Role> roles;
+                try {
+                    roles = rolesSpec.stream().map(s -> Role.valueOf((String) s)).collect(Collectors.toSet());
+                } catch (IllegalArgumentException e) {
+                    ResponseUtil.error(rc, 400, "invalid parameter: roles");
+                    return;
+                }
+                service.setRoles(roles);
+            }
+
+            if (siteId != null && siteId != 0) {
```

Comment:
I think we need to re-validate that the <service_name, site_id> combination remains unique

Reply:
Added the validation and a unit test

Reply:
sorry, I couldn't find the validation. Can you please add a link to it ?